### PR TITLE
[SYCL][HIP] Add additional AMD GPU targets

### PR DESCRIPTION
**Pull request by jinz2014**
The GFX targets are 909, 940, 941, 942, 1033, 1035, 1036, 1100, 1101, 1102, 1103, 1150, 1151, 1200, 1201.  

Thanks for your reviews.

### CODE REVIEW

File: clang/lib/Driver/ToolChains/SYCL.cpp [hunk 1]
```diff
@@ -932,6 +947,7 @@ SmallString<64> SYCL::gen::getGenDeviceMacro(StringRef DeviceName) {
                       .Case("gfx906", "AMD_GPU_GFX906")
                       .Case("gfx908", "AMD_GPU_GFX908")
                       .Case("gfx90a", "AMD_GPU_GFX90A")
+                      .Case("gfx90c", "AMD_GPU_GFX90C")
```
Comment (mdtoguchi) [inline clang/lib/Driver/ToolChains/SYCL.cpp:950]: Missing macro additions?

Comment (jinz2014) [reply clang/lib/Driver/ToolChains/SYCL.cpp:952]: Yes. Thanks.

File: sycl/doc/design/DeviceIf.md [hunk 2]
```diff
@@ -181,6 +197,7 @@ one of the following corresponding C++ macro names:
 * `__SYCL_TARGET_AMD_GPU_GFX906__`
 * `__SYCL_TARGET_AMD_GPU_GFX908__`
 * `__SYCL_TARGET_AMD_GPU_GFX90A__`
+* `__SYCL_TARGET_AMD_GPU_GFX90C__`
```
Comment (gmlueck) [inline sycl/doc/design/DeviceIf.md:200]: Should we add macros for the other new targets here too?

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc [hunk 3]
```diff
@@ -440,10 +456,30 @@ architecture.
...
+|AMD GCN GFX9 (Vega) architecture.
+
+|`amd_gpu_gfx941`
+|-
+|AMD GCN GFX9 (Vega) architecture.
+
+|`amd_gpu_gfx942`
+|-
+|AMD GCN GFX9 (Vega) architecture.
+
```
Comment (gmlueck) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc:482]: The description of all these new arch entries is the same.  Is that really right?  What's the difference between them?

File: sycl/doc/design/DeviceIf.md (hunk 2 above)
Comment (jinz2014) [reply sycl/doc/design/DeviceIf.md:201]: Yes. Thanks.

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc (hunk 3 above)
Comment (jinz2014) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc:482]: No. Vega and CDNA. Thanks.

**Review by steffenlarsen — APPROVED** [review]
Runtime changes and doc look good to me.

**Comment by steffenlarsen** [comment]
Tag @npmiller for awareness.

**Review by npmiller — APPROVED** [review]

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc [hunk 4]
```diff
@@ -472,10 +508,54 @@ architecture.
 |-
 |GCN GFX10.3 (RDNA 2) architecture.
 
+|`amd_gpu_gfx1033`
+|-
+|GCN GFX10.3 (RDNA 2) architecture.
```
Comment (gmlueck) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc:513]: These new entries are described as "GCN" while the previous ones are described as "AMD GCN".  Is that an important difference, or is it unintentional?

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc (hunk 3 above)
Comment (gmlueck) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc:482]: I don't really know these architectures, but many of the new entries have exactly the same description.  For example, `amd_gpu_gfx940`, `amd_gpu_gfx941`, and `amd_gpu_gfx942` are all described as "AMD GCN GFX9 (CDNA 3) architecture".

Is there really no difference between these three?  If that is the case, why do we have three enumerators?

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc (hunk 4 above)
Comment (jinz2014) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc:513]: I will add "AMD". Thanks.

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc (hunk 3 above)
Comment (jinz2014) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc:482]: Is it okay to group them ?
```
|`amd_gpu_gfx940`
|`amd_gpu_gfx941`
|`amd_gpu_gfx942`
|-
|AMD GCN GFX9 (CDNA 3) architecture.
```

Comment (gmlueck) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc:482]: This doesn't address my question.  If they are all exactly the same, why do we have three enumerators?  If they are different in some way, the description should explain the difference.

Comment (jinz2014) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc]: I updated the doc. If it still doesn't look good, please have a AMD product manager review the correctness of the names of the architectures. Thanks.

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc [hunk 5]
```diff
@@ -388,93 +404,57 @@ architecture.
...
+|AMD GCN 2.0 architecture.
 
-|`amd_gpu_gfx701`
+|`amd_gpu_gfx801`, `amd_gpu_gfx802`, `amd_gpu_gfx805`, `amd_gpu_gfx810`
 |-
-|AMD GCN GFX7 (Sea Islands (CI)) architecture.
+|AMD GCN 3.0 architecture.
 
-|`amd_gpu_gfx702`
+`amd_gpu_gfx803`
```
Comment (gmlueck) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc:415]: The table formatting is wrong from this point forward.

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc (hunk 3 above)
Comment (gmlueck) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc]: Thanks, @jinz2014.

**Review by gmlueck — APPROVED** [review]
spec changes OK

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc (hunk 5 above)
Comment (jinz2014) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc]: Thank you for your reviews!

**Comment by jinz2014** [comment]
@mdtoguchi, could you please approve the pull request if it looks right ? Thanks.

**Review by mdtoguchi — APPROVED** [review]

**Comment by againull** [comment]
Could you please merge latest sycl branch and resolve conflicts.

**Comment by jinz2014** [comment]
> @jinz2014 Could you please merge the latest sycl branch and resolve conflicts.

I applied the changes to the latest copy of the file.  Is that okay ?  Thanks.

**Comment by againull** [comment]
> > @jinz2014 Could you please merge the latest sycl branch and resolve conflicts.
> 
> I applied the changes to the latest copy of the file. Is that okay ? Thanks.

@jinz2014 Sorry, but unfortunately github show that there is still a conflict:
image

Did you push the changes you've made to the latest copy?

**Comment by jinz2014** [comment]
@againull   I am confused by the conflict report though I applied the changes to the latest file.

**Comment by againull** [comment]
> @againull I am confused by the conflict report though I applied the changes to the latest file.

No worries, conflict seemed trivial to me so I resolved in the github web interface. I will merge if testing results are fine.

**Pull request merged**